co-simulation/lib/mosaic-objects/src/main/java/org/eclipse/mosaic/lib/objects/detector/DetectorType.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright (C) 2023 LEIDOS.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package org.eclipse.mosaic.lib.objects.detector;
+
+
+
+import org.eclipse.mosaic.lib.objects.detector.gson.DetectorTypeAdapter;
+
+import com.google.gson.annotations.JsonAdapter;
+
+@JsonAdapter(DetectorTypeAdapter.class)
+public enum DetectorType {
+    SEMANTIC_LIDAR("SemanticLidar"),
+    RADAR("Radar"),
+    LIDAR("Lidar"),
+    SEMANTIC_SEGMENTATION_CAMERA("SemanticSegnmentationCamera"),
+    INSTANCE_SEGMENTATION_CAMERA("InstanceSegmentationCamera");
+ 
+    
+    
+    public final String label;
+    
+    /**
+     * Default constructor.
+     *
+     * @param label String
+     */
+    DetectorType(String label) {
+        this.label = label;
+    }
+    
+    /**
+     * Returns the enum mapped from an String name.
+     *
+     * @param label string.
+     * @return the enum mapped from String name.
+     */
+    public static DetectorType fromLabel(String label) {
+        for (DetectorType type: DetectorType.values()) {
+            if (type.label.equals(label)) {
```

Comment:
Maybe case insensitive comparison and remove any hidden leading or tailing spaces.

Reply:
I do not think that is necessary either. The goal here is not to allow any format of the label to be translated into the enumeration but rather to provide a method for the JSON adapter to read the enumeration from the json message.

Reply:
I am getting at is that the format including the case and spaces. Do we have a warning to tell why a label is not a valid DetectorType. EG: `SemanticLidar` is a valid DetectorType, but `semanticlidar` is not a valid DetectorType.

Reply:
I guess I can fix this.